akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -69,18 +69,34 @@ private[remote] object EnvelopeBuffer {
 /**
  * INTERNAL API
  */
-private[remote] trait LiteralCompressionTable {
+private[remote] trait LiteralCompression extends InboundCompression with OutboundCOmpression
 
-  def compressActorRef(ref: String): Int
+/**
+ * INTERNAL API
+ * Decompress and cause compression advertisements.
+ */
+private[remote] trait InboundCompression {
+  /** Incoming side counts how many times an ActorRef is used and then issues compression if it is considered "heavy" */
+  def hitActorRef(ref: String): Unit
   def decompressActorRef(idx: Int): String
```

Comment:
Duh, exactly what I was thinking but then fit into the existing interfaces.
All tables are ready to hold anything – will hold ActorRefs 👍